### ISSUE TITLE
Measure at arbitrary angles

### ISSUE THREAD

**Issue by Burtfaceman** [andrijac/ruler#64]
Hi. I like the app, but I'd like to be able to measure distances at arbitrary angles. The screenshot in the Microsoft Store shows this capability, but the app doesn't appear to have that functionality. How does one enable measurement at arbitrary angles?

Here's the screenshot from Microsoft Store:
![image](https://user-images.githubusercontent.com/81212159/165797102-e923b0a2-732b-4abf-b083-a9fd7afada0d.png)

This is what installed on my computer:
![image](https://user-images.githubusercontent.com/81212159/165797484-3402edd5-2991-460d-9878-7c0185517708.png)

**Comment by IsaacMorris1980** [comment]
@andrijac  i might be wrong but the picture from the store is a different ruler then yours correct?

**Comment by andrijac** [comment]
I would need help with this. @IsaacMorris1980 can you implement it?

**Comment by IsaacMorris1980** [comment]
I can try to implement it

**Comment by IsaacMorris1980** [comment]
@andrijac  Do we want this to be a command line paramater as well?

**Comment by andrijac** [comment]
@Burtfaceman yeah, this is different ruler, this ruler is not in Microsoft Store.

**Comment by IsaacMorris1980** [comment]
@andrijac  looks like i can not implement this using winforms the form itself will not rotate i can rotate the text but it is not usefull
  checking if wpf would allow it to be rotated

**Comment by IsaacMorris1980** [comment]
@andrijac  Having an issue rotating the wpf window as well still researching it.

**Comment by andrijac** [comment]
I suspected that it was hard to do... If it's not possible, leave it.

**Comment by IsaacMorris1980** [comment]
@andrijac the only potenial way is to make a headless full screen transparent window then draw the ruler as a movable control ontop of the window but it would block access to everything else on the screen. So i am leaving it unless it is acceptable to block access to the screen.

**Comment by GodSaveEarth** [comment]
sorry for necroposting, but +1 for custom angle ruler

**Comment by IsaacMorris1980** [comment]
Re-researching

**Comment by IsaacMorris1980** [comment]
@andrijac  i think i found a way to do this but it would change the project from winforms to a wpf project are you ok wth that

i can make a wpf page transparent allowing mouse clicks to access the desktop and host the winform ruler as a control  still researching angles

i will make a second repo to test that this will fully work

**Comment by andrijac** [comment]
@IsaacMorris1980 that would be complete rewrite of the app. You would need to implement all the same functionality in new app. If you think you have time to do it, go ahead, you can try in separate branch in same repo, just add new project to solution and give it a try there.

**Comment by IsaacMorris1980** [comment]
i will do that it will take time but i have that

**Comment by IsaacMorris1980** [comment]
i have rotation poc working however  no where near ready

**Comment by IsaacMorris1980** [comment]
This should be implemented in the build after next in the wpf ruler

**Comment by IsaacMorris1980** [comment]
going to be a little longer adding a few needed updates to the app before working on this
